File: 1413-minimum-value-to-get-positive-step-by-step-sum/1413-minimum-value-to-get-positive-step-by-step-sum.cpp
class Solution {
public:
    int minStartValue(vector<int>& nums) {
        int sum = 0;
        int minSum = INT_MAX;
        
        for (int i = 0; i < nums.size(); i++) {
            sum += nums[i];

            if (sum < minSum) {
                minSum = sum;
            }
        }
        
        return max(1, 1 - minSum);
    }
};

File: 1413-minimum-value-to-get-positive-step-by-step-sum/1413-minimum-value-to-get-positive-step-by-step-sum.java
class Solution {
    public int minStartValue(int[] nums) {
        // compute prefixSum and find the minimum value
        int total = 0;
        int min = Integer.MAX_VALUE;

        for (int num : nums) {
            total += num;

            if (total < min) {
                min = total;
            }
        }

        return Math.max(1, 1 - min);
    }
}